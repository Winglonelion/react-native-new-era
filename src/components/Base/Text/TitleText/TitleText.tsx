import React, { memo } from 'react';
import {
  ColorValue,
  StyleSheet,
  Text,
  TextStyle,
  StyleProp,
} from 'react-native';
import { FontFamily } from 'theme/CommonFonts';
import Colors from 'theme/colors';

export interface PropTypes {
  level?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  color?: ColorValue;
  style?: StyleProp<TextStyle>;
}

const LEVEL_STYLES: Record<string, TextStyle> = {
  h1: { fontSize: 24, lineHeight: 30, letterSpacing: 0.1, fontWeight: '500' },
  h2: { fontSize: 18, lineHeight: 24, letterSpacing: 0.1, fontWeight: '500' },
  h3: { fontSize: 16, lineHeight: 22, letterSpacing: 0.1, fontWeight: '500' },
  h4: { fontSize: 14, lineHeight: 20, letterSpacing: 0.1, fontWeight: '500' },
  h5: { fontSize: 12, lineHeight: 18, letterSpacing: 0.1, fontWeight: '500' },
  h6: { fontSize: 10, lineHeight: 16, letterSpacing: 0.1, fontWeight: '500' },
};

const TitleText: React.FC<PropTypes> = memo(
  ({ children, level = 'h4', style, color }) => {
    return (
      <Text
        style={[
          styles.text,
          LEVEL_STYLES[level],
          color ? { color } : undefined,
          style,
        ]}>
        {children}
      </Text>
    );
  },
);

const styles = StyleSheet.create({
  text: {
    fontSize: 14,
    color: Colors.black,
    fontFamily: FontFamily.Helvetica,
  },
});
export default TitleText;
