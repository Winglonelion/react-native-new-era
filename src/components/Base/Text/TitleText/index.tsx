import TitleText, { PropTypes } from './TitleText';

export type TitleTextPropTypes = PropTypes;
export default TitleText;
