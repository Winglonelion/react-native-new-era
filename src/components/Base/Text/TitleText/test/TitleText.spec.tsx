/* eslint-disable react-native/no-raw-text */
import React from 'react';
import { shallow } from 'enzyme';
import TitleText, { PropTypes } from '../TitleText';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<TitleText {...props}>{'TEST'}</TitleText>);
  return {
    props,
    wrapper,
  };
};

describe('TitleText check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({
      level: 'h1',
      color: '#EEE',
      style: { lineHeight: 30 },
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot without props', () => {
    const { wrapper } = setup({});
    expect(wrapper).toMatchSnapshot();
  });
});
