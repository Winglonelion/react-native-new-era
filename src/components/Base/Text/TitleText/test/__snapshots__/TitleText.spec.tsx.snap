// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleText check snapshot should match snapshot with props 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Helvetica Neue",
        "fontSize": 14,
      },
      Object {
        "fontSize": 24,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 30,
      },
      Object {
        "color": "#EEE",
      },
      Object {
        "lineHeight": 30,
      },
    ]
  }
>
  TEST
</Text>
`;

exports[`TitleText check snapshot should match snapshot without props 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Helvetica Neue",
        "fontSize": 14,
      },
      Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 20,
      },
      undefined,
      undefined,
    ]
  }
>
  TEST
</Text>
`;
